src/OmniSharp.Script/ScriptProjectSystem.cs

```diff
@@ -121,18 +144,38 @@ public void Initalize(IConfiguration configuration)
 
             var commonReferences = new HashSet<MetadataReference>();
 
-            // if we have no context, then we also have no dependencies
-            // we can assume desktop framework
-            // and add mscorlib
+            
             if (runtimeContexts == null || runtimeContexts.Any() == false)
             {
-                _logger.LogInformation("Unable to find project context for CSX files. Will default to non-context usage.");
+                _logger.LogInformation($"Unable to find project context for CSX files. Will default to non-context usage for target framework {_targetFrameWork.Value}");
+
+                // We are running in the context of a .Net Core app.
+                if (_targetFrameWork.Value.StartsWith(".NETCoreApp"))
+                {                    
+                    var runtimeId = RuntimeEnvironment.GetRuntimeIdentifier();
+                    var inheritedAssemblyNames = DependencyContext.Default.GetRuntimeAssemblyNames(runtimeId).Where(x =>
+                        x.FullName.StartsWith("system.", StringComparison.OrdinalIgnoreCase) ||
+                        x.FullName.StartsWith("microsoft.codeanalysis", StringComparison.OrdinalIgnoreCase) ||
+                        x.FullName.StartsWith("mscorlib", StringComparison.OrdinalIgnoreCase));
+
+                    foreach (var inheritedAssemblyName in inheritedAssemblyNames)
+                    {
+                        var assembly = Assembly.Load(inheritedAssemblyName);
```

Comment:
We generally use an [`IAssemblyLoader`](https://github.com/OmniSharp/omnisharp-roslyn/blob/dev/src/OmniSharp.Abstractions/Services/IAssemblyLoader.cs) so that we can track assemblies that are loaded at runtime. Just import `IAssemblyLoader` in the constructor and use it here.